File: baekjoon/9012/Lactea98/9012.c
#include <stdio.h>
#include <string.h>

int main() {
    int num;
    scanf("%d", &num);

    for(int i=0; i<num; i++){
        char input[50];
        scanf("%s", input);

        int length = strlen(input);
        int stack = 0;
        int check = 0;
        for(int i=0; i<length; i++){
            if(input[i] == '('){
                stack++;
            }
            else{
                if(stack == 0){
                    check = 1;
                    break;
                }
                else stack--;
            }
        }

        if(!check && stack == 0) printf("YES\n");
        else printf("NO\n");
    }
}

File: baekjoon/9012/Lactea98/9012.py
num = int(input())

result = []
for i in range(num):
    data = []
    check = True

    input_data = input()

    for d in input_data:
        if d == ")":
            if len(data) == 0 or data[len(data) - 1] != "(":
                check = False
                break
            else:
                data.pop(len(data) - 1)
        else :
            data.append(d)
    
    if check == False or len(data) != 0:
        result.append("NO")
    else:
        result.append("YES")


print("\n".join(result))